### PR TITLE
Add a function to extend scheme's color array

### PR DESCRIPTION
**Pull request by JohnArcher**
You can add a new option called `custom` which should be a function that receives a clone of the configured scheme and has to return an array of colors.
If no or an empty array is returned the original scheme is used.

### CODE REVIEW

File: src/plugins/plugin.colorschemes.js [hunk 1]
```diff
@@ -20,16 +20,32 @@ export default {
...
-			length = scheme.length;
+
 			if (scheme) {
+
+				// clone the original scheme
+				schemeClone = scheme.slice();
+
+				if (options.custom) {
+					// Execute own custom color function
+					var colorFunctionResult = options.custom(schemeClone);
```
Comment (nagix) [inline src/plugins/plugin.colorschemes.js]: The utility function `helpers.callback` is available in Chart.js, so let's use it. This function checks if `options.custom ` is defined and it is a function, and returns `undefined` if not. So, we don't need `if (options.custom)` in the previous line. 
```js
var colorFunctionResult = helpers.callback(options.custom, [schemeClone]);
```

File: README.md [hunk 2]
```diff
@@ -77,6 +77,34 @@ All available options are listed below.
 | `fillAlpha` | `Number` | `0.5` | The transparency value for the line fill color. Must be a number between `0.0` (fully transparent) and `1.0` (no transparency).
 | `scheme` | `String` | `'brewer.Paired12'` | Color scheme name that is any of [Color Chart](https://nagix.github.io/chartjs-plugin-colorschemes/colorchart.html).
 | `reverse` | `Boolean` | `false` | If set to `true`, the order of the colors in the selected  scheme is reversed.
+| `custom` | `Function` | undefined | A function that takes a copy of `scheme` in order to extend the predefined scheme colors. [more...](#custom-function).
```
Comment (nagix) [inline README.md]: Can you replace ``A function that takes a copy of `scheme` `` with ``A function that takes a copy of the color string array for `scheme` ``?

File: src/plugins/plugin.colorschemes.js (hunk 1 above)
Comment (JohnArcher) [reply src/plugins/plugin.colorschemes.js]: Great, didn't know that utility function. I added it.

File: README.md (hunk 2 above)
Comment (JohnArcher) [reply README.md]: Done.

File: src/plugins/plugin.colorschemes.js [hunk 3]
```diff
@@ -20,16 +20,30 @@ export default {
...
 			if (scheme) {
+
+				// clone the original scheme
+				schemeClone = scheme.slice();
+
+				// Execute own custom color function
+				var colorFunctionResult = helpers.callback(options.custom, [schemeClone]);
+
+				// check if we really received a filled array; otherwise we keep and use the originally cloned scheme
+				if (colorFunctionResult instanceof Array && colorFunctionResult.length > 0) {
```
Comment (nagix) [inline src/plugins/plugin.colorschemes.js]: I prefer `helpers.isArray(colorFunctionResult)` to `colorFunctionResult instanceof Array`.

File: README.md [hunk 4]
```diff
@@ -77,6 +77,34 @@ All available options are listed below.
 | `fillAlpha` | `Number` | `0.5` | The transparency value for the line fill color. Must be a number between `0.0` (fully transparent) and `1.0` (no transparency).
 | `scheme` | `String` | `'brewer.Paired12'` | Color scheme name that is any of [Color Chart](https://nagix.github.io/chartjs-plugin-colorschemes/colorchart.html).
 | `reverse` | `Boolean` | `false` | If set to `true`, the order of the colors in the selected  scheme is reversed.
+| `custom` | `Function` | undefined | A function that takes a copy of the color string array for `scheme` in order to extend the predefined scheme colors. [more...](#custom-function).
```
Comment (nagix) [inline README.md]: nit: Use `` `undefined` `` instead of `undefined`.

File: src/plugins/plugin.colorschemes.js (hunk 3 above)
Comment (JohnArcher) [reply src/plugins/plugin.colorschemes.js]: Done.

File: README.md (hunk 4 above)
Comment (JohnArcher) [reply README.md]: Done.

**Comment by nagix** [comment]
@JohnArcher Thanks for your contribution!

**Comment by JohnArcher** [comment]
@nagix Thanks for your professional help and accepting this contribution. It was a pleasure!

**Pull request merged**